### PR TITLE
Feat matrix input

### PR DESCRIPTION
**Pull request by Fabianstw**
Component s.t. the user can enter an answer inside a matrix / vector. 

- changed FromInputField slightly, so the align option is now a type option, for special classes for the input 

Example, how it currently looks like 👍 
Screenshot 2024-10-22 at 15 51 21

### CODE REVIEW

File: front-end/src/components/MatrixInput.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+ * @param matrixObject
+ */
+export function MatrixInput({ matrixObject }: { matrixObject: string }): ReactElement {
+  const parsedMatrixObject = JSON.parse(matrixObject) as MatrixInputProps
+  const rows = parsedMatrixObject.rows
+  const cols = parsedMatrixObject.cols
+
+  const { theme } = useTheme()
+
+  const divHeight = 80 + (rows - 2) * 45
```
Comment (AbcAeffchen) [inline front-end/src/components/MatrixInput.tsx:36]: These magic numbers depend on the height of the input fields. I assume they are inferred by trial and error. At least I couldn't find the height definitions of the input fields or they are defined as `h-8` or something like that. Maybe there is a good html/css solution?

**Review by AbcAeffchen — COMMENTED** [review]
Looks good. I assume the parenthesis scales with the matrix size.

File: front-end/src/components/MatrixInput.tsx (hunk 1 above)
Comment (Fabianstw) [reply front-end/src/components/MatrixInput.tsx:36]: I took a look again and using the web inspector it seems like each field (with padding) is 42px. So I would change this to `const divHeight = rows * 42`. 
tested it for different values from 1 to 22 and looks good for those. Its still a _magic number_ but not trail and error anymore

**Comment by AbcAeffchen** [comment]
From my side, this can be merged now.

**Pull request merged**